### PR TITLE
fix: bump transformers lower bound to 5.5.4 to exclude broken Qwen2.5-VL RoPE releases

### LINKED ISSUE #3648: Qwen2.5-VL-based policy performance degradation caused by RoPE index calculation bugs in Transformers v5.4.0
### Ticket Type

🐛 Bug Report (Something isn't working)

### Environment & System Info

Hardware and host runtime:

- Host OS: `Linux 3.10.0-957.el7.x86_64 #1 SMP Thu Nov 8 23:39:32 UTC 2018`
- Slurm node: `x86_64`, `2` sockets, `32` cores/socket, `2` threads/core, `128` logical CPUs, `2060000M` real memory, `gpu:8`
- CPU: `Intel(R) Xeon(R) Platinum 8358P CPU @ 2.60GHz`
- GPU: `8 x NVIDIA A800-SXM4-80GB` per node
- GPU driver: `550.90.07`, CUDA runtime: `12.4`

Container and Python runtime:

- Container OS: Debian GNU/Linux 12 (`bookworm`)
- Python: `3.12.13`

<details>
<summary>Key package versions</summary>

| Package | Version |
| --- | --- |
| `torch` | `2.7.0+cu118` |
| PyTorch CUDA build | `11.8` |
| cuDNN reported by PyTorch | `90100` |
| `torchvision` | `0.22.0+cu118` |
| `transformers` | `5.4.0` |
| `triton` | `3.3.0` |
| `pytest` | `8.4.2` |
| `accelerate` | `1.13.0` |
| `datasets` | `4.8.4` |
| `huggingface_hub` | `1.10.2` |
| `safetensors` | `0.7.0` |
| `tokenizers` | `0.22.2` |
| `numpy` | `2.2.6` |
| `scipy` | `1.17.1` |
| `pillow` | `12.2.0` |
| `diffusers` | `0.35.2` |
| `opencv-python` | `4.13.0.92` |
| `einops` | `0.8.2` |

</details>

### Description

LeRobot's current `transformers-dep` lower bound allows `transformers==5.4.0`, which contains a Qwen2.5-VL still-image RoPE index bug: still images can be routed through the video temporal-scaling path in `get_rope_index`, so image-token temporal positions are multiplied by `tokens_per_second` while height/width positions remain unscaled.

This silently changes Qwen2.5-VL image-token RoPE phases and can severely degrade Qwen2.5-VL-based robot policies such as `wallx` and `eo1`.

Policy-level regression of `eo1` observed on Libero (average success drops by `23.30625` points):

| code version | Libero spatial | Libero object | Libero goal | Libero 10 | average |
| --- | ---: | ---: | ---: | ---: | ---: |
| `transformers 4.56.0` | `100.0` | `98.8` | `98.1` | `90.6` | `96.9` |
| `transformers 5.4.0` | `55.0` | `65.0` | `86.875` | `87.50` | `73.59375` |

This matches Transformers issue [#45325](https://github.com/huggingface/transformers/issues/45325). The upstream fix is PR [#45330](https://github.com/huggingface/transformers/pull/45330), released in Transformers [v5.5.4](https://github.com/huggingface/transformers/releases/tag/v5.5.4).

The vulnerable v5.4.0 code applies temporal scaling to both images and videos:

```python
time_interval = tokens_per_second * int(next(second_per_grid_ts))
```

The fixed code applies that scaling only to videos:

```python
if modality_type == 2:
    time_interval = tokens_per_second * int(next(second_per_grid_ts))
else:
    time_interval = 1
```

For VLA policies, this matters because robot observations are usually still camera frames, not videos. In a local EO1 trace with two camera images, we control the raw data, processor outputs, image tensors, token layout, and action span, but the image-token temporal axis drifts:

| image span | token span | expected temporal origin | buggy temporal origin | height/width origin |
| --- | ---: | ---: | ---: | ---: |
| first image | `[16, 97)` | `16` | `32` | `16` |
| second image | `[99, 180)` | `27` | `54` | `27` |

Local trace facts:

- `position_ids` shape is `[3, 8, 221]`.
- `diff_count = 1296`.
- Differences appear only on axis `0`, the temporal/text axis.
- Height and width axes are identical.

The canonical Qwen forward path also produces the same shifted still-image RoPE semantics:

| comparison | max_abs | mean_abs |
| --- | ---: | ---: |
| old `position_ids` vs new `compute_3d_position_ids()` | `27.0` | `5.2534` |
| old `position_ids` vs new native `get_rope_index()` | `27.0` | `5.2534` |
| new `compute_3d_position_ids()` vs new native `get_rope_index()` | `0` | `0` |

After the wrong RoPE positions are consumed by attention, the drift accumulates through the Qwen decoder and reaches the action head:

| tensor | max_abs | mean_abs |
| --- | ---: | ---: |
| `decoder_layer_00/input_hidden_states` | `0.5` | `0.00941` |
| `decoder_layer_00/output_hidden_states` | `2.3125` | `0.09074` |
| `decoder_layer_18/output_hidden_states` | `192.0` | `0.29540` |
| `decoder_layer_35/output_hidden_states` | `488.0` | `1.29809` |
| `qwen_forward_00/forward/last_hidden_state` | `36.53125` | `0.29315` |
| `eo1/action_out_proj_00/forward/output` | `2.1573` | `0.06862` |
| `forward/loss` | `0.19506` | `0.19506` |

> [!NOTE]
> In short: image tokens are present and spatially aligned, but their temporal RoPE phase is wrong. The error is then amplified by attention and changes action prediction.

### Context & Reproduction

I ran the same minimal script twice in the same container: once with `transformers==5.4.0`, and once with `transformers==5.5.4`. The check is CPU-only and does not instantiate model weights; it only loads the Qwen2.5-VL processor/config and calls `get_rope_index`.

Command:

```bash
python repro_qwen2_5_vl_rope_index.py
```

<details>
<summary>repro_qwen2_5_vl_rope_index.py</summary>

```python
import os

import numpy as np
from PIL import Image
import torch
from transformers import Qwen2_5_VLConfig, Qwen2_5_VLProcessor, __version__ as transformers_version
from transformers.models.qwen2_5_vl.modeling_qwen2_5_vl import Qwen2_5_VLModel


class DummyQwen2_5VLModel:
    get_rope_index = Qwen2_5_VLModel.get_rope_index
    get_vision_position_ids = Qwen2_5_VLModel.get_vision_position_ids

    def __init__(self, config):
        self.config = config


def contiguous_spans(index):
    breaks = (index[1:] != index[:-1] + 1).nonzero().flatten() + 1
    chunks = torch.tensor_split(index, breaks.tolist())
    return [(int(chunk[0]), int(chunk[-1]) + 1) for chunk in chunks if len(chunk)]


model_id = os.environ.get("QWEN25VL_MODEL", "Qwen/Qwen2.5-VL-3B-Instruct")
processor = Qwen2_5_VLProcessor.from_pretrained(model_id)
config = Qwen2_5_VLConfig.from_pretrained(model_id)
model = DummyQwen2_5VLModel(config)

image = Image.fromarray(np.zeros((448, 448, 3), dtype=np.uint8))
prompt = (
    "<|im_start|>user\n"
    "<|vision_start|><|image_pad|><|vision_end|>"
    "<|vision_start|><|image_pad|><|vision_end|>"
    " predict the robot action<|im_end|>\n"
    "<|im_start|>assistant\n"
)
inputs = processor(
    text=[prompt],
    images=[image, image],
    return_tensors="pt",
    return_mm_token_type_ids=True,
)

# `get_rope_index` accepts `**kwargs`, so processor fields unrelated to RoPE
# such as `pixel_values` are ignored.
position_ids, _ = model.get_rope_index(**inputs)
image_positions = (inputs["mm_token_type_ids"][0] == 1).nonzero(as_tuple=False).flatten()

print(f"transformers={transformers_version}, tokens_per_second={config.vision_config.tokens_per_second}")
for span_id, (start, end) in enumerate(contiguous_spans(image_positions)):
    temporal = int(position_ids[0, 0, start])
    height = int(position_ids[1, 0, start])
    width = int(position_ids[2, 0, start])
    print(
        f"image_span={span_id}, token_span=[{start},{end}), "
        f"temporal={temporal}, height={height}, width={width}, "
        f"aligned={temporal == height == width}"
    )
```

</details>

Expected invariant for still images:

```python
temporal_start == height_start == width_start
```

### Relevant logs or stack trace

The same two-image still-frame prompt gives different RoPE starts:

| transformers | tokens_per_second | image_span | token_span | temporal_start | height_start | width_start | aligned |
| --- | ---: | ---: | --- | ---: | ---: | ---: | --- |
| `5.4.0` | `2` | `0` | `[4,260)` | `8` | `4` | `4` | no |
| `5.4.0` | `2` | `1` | `[262,518)` | `44` | `22` | `22` | no |
| `5.5.4` | `2` | `0` | `[4,260)` | `4` | `4` | `4` | yes |
| `5.5.4` | `2` | `1` | `[262,518)` | `22` | `22` | `22` | yes |

Summary:

- In `transformers==5.4.0`, still-image `temporal_start` is scaled by `tokens_per_second=2`.
- In `transformers==5.5.4`, `temporal_start`, `height_start`, and `width_start` are aligned for both still-image spans.

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [x] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.

### Additional Info / Workarounds

Possible fix options:

- Move LeRobot's `transformers-dep` bound to `"transformers>=5.4.0,<5.6.0"`
- Add an explicit Qwen2.5-VL RoPE-index compatibility monkey patch in Qwen2.5-VL-based policies

Affected policy scope:

- `wallx` depends on `lerobot[transformers-dep]` and `lerobot[qwen-vl-utils-dep]`.
- `eo1` depends on `lerobot[transformers-dep]` and `lerobot[qwen-vl-utils-dep]`.

Related prs / issues:

- Transformers issue: <https://github.com/huggingface/transformers/issues/45325>
- Transformers PR: <https://github.com/huggingface/transformers/pull/45330>
- Transformers v5.5.4 release: <https://github.com/huggingface/transformers/releases/tag/v5.5.4>

### PR DESCRIPTION
**Pull request by yushuosun**
## Summary / Motivation

Closes #3648.

The `transformers-dep` extra currently allows `transformers>=5.4.0,<5.6.0`.
Releases 5.4.0–5.5.3 contain a bug in
`Qwen2_5_VLForConditionalGeneration.get_rope_index` where still images are
incorrectly routed through the video temporal-scaling path. Concretely,
the variable `second_per_grid_t` is set to a non-zero value for image
tokens, causing `temporal_start` to be multiplied by `tokens_per_second`
rather than remaining zero-scaled. The result is that every image token
receives a shifted temporal position encoding, degrading policy performance
on all Qwen2.5-VL-backed policies.

The bug was fixed in transformers 5.5.4 via
huggingface/transformers#45330 (tracking issue
huggingface/transformers#45325).

## Root cause

`lerobot/policies/eo1/modeling_eo1.py` (line 566) calls
`vlm_backbone.model.get_rope_index(...)` directly through the transformers
public API. Any environment that resolves `transformers>=5.4.0` to a
release between 5.4.0 and 5.5.3 (inclusive) will silently receive wrong
position IDs for still-image observations.

`wall_x` maintains its own local copy of `get_rope_index` and already sets
`second_per_grid_t = 0` for images unconditionally, so it is not affected
by this dependency range.

## What changed

The `transformers-dep` extra in `pyproject.toml`:
```diff
-transformers-dep = ["transformers>=5.4.0,<5.6.0"]
+transformers-dep = ["transformers>=5.5.4,<5.6.0"]
```

The upper bound is unchanged; only the lower bound is raised to exclude the
broken 5.4.0–5.5.3 range. All extras that pull in `transformers-dep`
(`eo1`, `wallx`, `sarm`, `smolvla`, `pi`, `hilserl`, `xvla`, `peft`,
`libero`, etc.) inherit this tighter bound automatically.

## Duplicate check

- **Track A – open PRs referencing #3648**: no existing PR found citing this issue.
- **Track B – issue comments**: issue #3648 has 0 comments; no in-progress fix exists.

### CODE REVIEW

**Comment by yushuosun** [comment]
Closing because the dependency set on main has moved substantially, this branch now conflicts, and a version-bound-only change without a current regression test is no longer a good patch. I can revisit with a reproducer if the incompatibility returns.
